Failing tests on RDS

- EdgeDB Version: 157e58b78315339ff620e04d7afcecb05f0505df
- OS Version: macOS 11.2.1 Darwin Kernel Version 20.3.0 xnu-7195.81.3~1/RELEASE_X86_64
- AWS RDS Version: 12.4

This is an issue to track all the failing tests when running on AWS RDS.

- [x] `ALTER ROLE ... SET password := ...` fails. Fixed in #2282 
- [x] test_edgeql_ddl_role_02 (test_edgeql_ddl.TestEdgeQLDDL)

   ```
   Traceback (most recent call last):
     File "edb/testbase/server.py", line 127, in wrapper
       self.loop.run_until_complete(
     File "asyncio/base_events.py", line 642, in run_until_complete
       return future.result()
     File "tests/test_edgeql_ddl.py", line 6284, in test_edgeql_ddl_role_02
       await self.assert_query_result(
     File "edb/testbase/server.py", line 534, in assert_query_result
       self._assert_data_shape(res, exp_result_json, message=msg)
     File "edb/testbase/server.py", line 742, in _assert_data_shape
       return _assert_generic_shape((), data, shape)
     File "edb/testbase/server.py", line 710, in _assert_generic_shape
       return _assert_list_shape(path, data, shape)
     File "edb/testbase/server.py", line 660, in _assert_list_shape
       _assert_generic_shape(
     File "edb/testbase/server.py", line 714, in _assert_generic_shape
       return _assert_dict_shape(path, data, shape)
     File "edb/testbase/server.py", line 619, in _assert_dict_shape
       _assert_generic_shape(path + (f'["{sk}"]',), data[sk], sv)
     File "edb/testbase/server.py", line 730, in _assert_generic_shape
       self.fail(
   AssertionError: data shape differs: False != True PATH: [0]["superuser"]
   ```

- [x] test_server_proto_ddlprop_01 (test_server_proto.TestServerProtoDdlPropagation)

   ```
   Traceback (most recent call last):
     File "edb/testbase/server.py", line 127, in wrapper
       self.loop.run_until_complete(
     File "asyncio/base_events.py", line 642, in run_until_complete
       return future.result()
     File "tests/test_server_proto.py", line 1895, in test_server_proto_ddlprop_01
       async with tb.start_edgedb_server(adjacent_to=self.con) as sd:
     File "edb/testbase/server.py", line 1495, in __aenter__
       data = await asyncio.wait_for(
     File "asyncio/tasks.py", line 481, in wait_for
       return fut.result()
     File "edb/testbase/server.py", line 1431, in _read_runtime_info
       raise RuntimeError("EdgeDB server terminated")
   RuntimeError: EdgeDB server terminated
   ```

- [x] test_edgeql_select_polymorphic_10 (test_edgeql_select.TestEdgeQLSelect) (passes when ran separately)

   ```
   Traceback (most recent call last):
     File "edb/testbase/server.py", line 127, in wrapper
       self.loop.run_until_complete(
     File "asyncio/base_events.py", line 642, in run_until_complete
       return future.result()
     File "tests/test_edgeql_select.py", line 1538, in test_edgeql_select_polymorphic_10
       await self.assert_query_result(
     File "edb/testbase/server.py", line 547, in assert_query_result
       res = await self.con._fetchall(
     File "edgedb/asyncio_con.py", line 201, in _fetchall
       result, _ = await self._impl._protocol.execute_anonymous(
     File "edgedb/protocol/protocol.pyx", line 633, in execute_anonymous
     File "edgedb/protocol/protocol.pyx", line 405, in _execute
   edgedb.errors.InternalServerError: remaining connection slots are reserved for non-replication superuser connections
   ```

- [x] Tests hangs in GitHub Action (fixed by adding TCP keepalive)
- [x] test_server_auth_01 (test_server_auth.TestServerAuth)

  ```
  Traceback (most recent call last):
    File "/home/runner/work/edgedb/edgedb/edb/testbase/server.py", line 127, in wrapper
      self.loop.run_until_complete(
    File "/opt/hostedtoolcache/Python/3.9.2/x64/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
      return future.result()
    File "/home/runner/work/edgedb/edgedb/tests/test_server_auth.py", line 40, in test_server_auth_01
      await self.connect(
    File "/opt/hostedtoolcache/Python/3.9.2/x64/lib/python3.9/contextlib.py", line 124, in __exit__
      next(self.gen)
    File "/home/runner/work/edgedb/edgedb/edb/testbase/server.py", line 215, in assertRaisesRegex
      raise
  AssertionError: AuthenticationError not raised
  ```

- [x] test_server_proto_concurrent_global_ddl (test_server_proto.TestServerProtoConcurrentGlobalDDL)

  ```
  Traceback (most recent call last):
    File "/home/runner/work/edgedb/edgedb/tests/test_server_proto.py", line 2852, in test_server_proto_concurrent_global_ddl
      g.create_task(con.execute(f'''
    File "/home/runner/work/edgedb/edgedb/edb/common/taskgroup.py", line 170, in __aexit__
      raise me from None
  edb.common.taskgroup.TaskGroupError: unhandled errors in a TaskGroup; 1 sub errors: (SchemaError)
   + SchemaError: role 'concurrent_1' is already present in the schema
   |   File "/home/runner/.virtualenvs/.venv/lib/python3.9/site-packages/edgedb/asyncio_con.py", line 359, in execute
   |     await self._impl._protocol.simple_query(
   |   File "edgedb/protocol/protocol.pyx", line 578, in simple_query
  
  
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/home/runner/work/edgedb/edgedb/edb/testbase/server.py", line 127, in wrapper
      self.loop.run_until_complete(
    File "/opt/hostedtoolcache/Python/3.9.2/x64/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
      return future.result()
    File "/home/runner/work/edgedb/edgedb/tests/test_server_proto.py", line 2856, in test_server_proto_concurrent_global_ddl
      self.assertIn(
  AssertionError: <class 'edgedb.errors.TransactionSerializationError'> not found in {<class 'edgedb.errors.SchemaError'>}
  ```

- [x] test_server_proto_configure_describe_database_config (test_server_config.TestServerConfig)

  ```
  Traceback (most recent call last):
    File "/home/runner/work/edgedb/edgedb/edb/testbase/server.py", line 127, in wrapper
      self.loop.run_until_complete(
    File "/opt/hostedtoolcache/Python/3.9.2/x64/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
      return future.result()
    File "/home/runner/work/edgedb/edgedb/tests/test_server_config.py", line 953, in test_server_proto_configure_describe_database_config
      self.assertIn(conf1, res)
  AssertionError: "CONFIGURE CURRENT DATABASE SET singleprop := '1337';" not found in ''
  ```

- [x] test_server_proto_configure_describe_system_config (test_server_config.TestServerConfig)

  ```
  Traceback (most recent call last):
    File "/home/runner/work/edgedb/edgedb/edb/testbase/server.py", line 127, in wrapper
      self.loop.run_until_complete(
    File "/opt/hostedtoolcache/Python/3.9.2/x64/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
      return future.result()
    File "/home/runner/work/edgedb/edgedb/tests/test_server_config.py", line 928, in test_server_proto_configure_describe_system_config
      self.assertIn(conf1, res)
  AssertionError: "CONFIGURE SYSTEM SET singleprop := '1337';" not found in "CONFIGURE SYSTEM INSERT cfg::Auth {\n    priority := 0,\n    user := {'*'},\n    method := (\n        (INSERT cfg::Trust)\n    ),\n};\n"
  ```

--- a/.github/aws-rds/main.tf
+++ b/.github/aws-rds/main.tf
@@ -1,9 +1,8 @@
 resource "aws_db_instance" "default" {
   allocated_storage = 10
   engine = "postgres"
-  engine_version = "12.5"
-  instance_class = "db.t3.small"
-//  instance_class = "db.m6g.large"
+  engine_version = "12.8"
+  instance_class = "db.m6g.large"
   name = "edbtest"
   username = "edbtest"
   password = var.password

--- a/.github/do-database/main.tf
+++ b/.github/do-database/main.tf
@@ -18,7 +18,7 @@ resource "digitalocean_database_cluster" "default" {
   name = "edbtest"
   engine     = "pg"
   version    = "12"
-  size       = "db-s-1vcpu-2gb"
+  size       = "db-s-4vcpu-8gb"
   region     = "nyc1"
   node_count = 1
 }

--- a/.github/do-database/outputs.tf
+++ b/.github/do-database/outputs.tf
@@ -12,6 +12,7 @@ output "db_instance_user" {
 
 output "db_instance_password" {
   value = digitalocean_database_cluster.default.password
+  sensitive = true
 }
 
 output "db_instance_database" {

--- a/.github/gcp-cloud-sql/main.tf
+++ b/.github/gcp-cloud-sql/main.tf
@@ -9,7 +9,7 @@ resource "google_sql_database_instance" "default" {
   deletion_protection = false
 
   settings {
-    tier = "db-g1-small"
+    tier = "db-custom-1-3840"
 
     ip_configuration {
       authorized_networks {

--- a/.github/workflows.src/tests-managed-pg.tpl.yml
+++ b/.github/workflows.src/tests-managed-pg.tpl.yml
@@ -84,7 +84,7 @@ jobs:
       env:
         EDGEDB_TEST_BACKEND_DSN: postgres://edbtest:${{ secrets.AWS_RDS_PASSWORD }}@${{ needs.setup-aws-rds.outputs.pghost }}/postgres
       run: |
-        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
+        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN --testmode
         edb test -j2 -v --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
 
   teardown-aws-rds:
@@ -196,7 +196,7 @@ jobs:
       env:
         EDGEDB_TEST_BACKEND_DSN: postgres://${{ steps.pguser.outputs.stdout }}:${{ steps.pgpass.outputs.stdout }}@${{ steps.pghost.outputs.stdout }}:${{ steps.pgport.outputs.stdout }}/${{ steps.pgdatabase.outputs.stdout }}
       run: |
-        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
+        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN --testmode
         edb test -j2 -v --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
 
   teardown-do-database:
@@ -271,7 +271,7 @@ jobs:
       env:
         EDGEDB_TEST_BACKEND_DSN: postgres://postgres:${{ secrets.AWS_RDS_PASSWORD }}@${{ needs.setup-gcp-cloud-sql.outputs.pghost }}/postgres
       run: |
-        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
+        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN --testmode
         edb test -j2 -v --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
 
   teardown-gcp-cloud-sql:

--- a/.github/workflows.src/tests-managed-pg.tpl.yml
+++ b/.github/workflows.src/tests-managed-pg.tpl.yml
@@ -21,7 +21,7 @@ name: Tests on Managed PostgreSQL
 
 on:
   schedule:
-    - cron: "0 3 * * *"
+    - cron: "0 3 * * 6"
   workflow_dispatch:
     inputs: {}
   push:

--- a/.github/workflows/tests-managed-pg.yml
+++ b/.github/workflows/tests-managed-pg.yml
@@ -2,7 +2,7 @@ name: Tests on Managed PostgreSQL
 
 on:
   schedule:
-    - cron: "0 3 * * *"
+    - cron: "0 3 * * 6"
   workflow_dispatch:
     inputs: {}
   push:

--- a/.github/workflows/tests-managed-pg.yml
+++ b/.github/workflows/tests-managed-pg.yml
@@ -462,7 +462,7 @@ jobs:
       env:
         EDGEDB_TEST_BACKEND_DSN: postgres://edbtest:${{ secrets.AWS_RDS_PASSWORD }}@${{ needs.setup-aws-rds.outputs.pghost }}/postgres
       run: |
-        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
+        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN --testmode
         edb test -j2 -v --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
 
   teardown-aws-rds:
@@ -706,7 +706,7 @@ jobs:
       env:
         EDGEDB_TEST_BACKEND_DSN: postgres://${{ steps.pguser.outputs.stdout }}:${{ steps.pgpass.outputs.stdout }}@${{ steps.pghost.outputs.stdout }}:${{ steps.pgport.outputs.stdout }}/${{ steps.pgdatabase.outputs.stdout }}
       run: |
-        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
+        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN --testmode
         edb test -j2 -v --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
 
   teardown-do-database:
@@ -912,7 +912,7 @@ jobs:
       env:
         EDGEDB_TEST_BACKEND_DSN: postgres://postgres:${{ secrets.AWS_RDS_PASSWORD }}@${{ needs.setup-gcp-cloud-sql.outputs.pghost }}/postgres
       run: |
-        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
+        edb server --bootstrap-only --backend-dsn=$EDGEDB_TEST_BACKEND_DSN --testmode
         edb test -j2 -v --backend-dsn=$EDGEDB_TEST_BACKEND_DSN
 
   teardown-gcp-cloud-sql:

